homeassistant/components/overkiz/climate_entities/valve_heating_temperature_interface.py

```diff
@@ -0,0 +1,137 @@
+
+PRESET_MANUAL = "manual"
+PRESET_FROST_PROTECTION = "frost_protection"
+
+OVERKIZ_TO_HVAC_ACTION: dict[str, HVACAction] = {
+    OverkizCommandParam.OPEN: HVACAction.HEATING,
+    OverkizCommandParam.CLOSED: HVACAction.IDLE,
+}
+
+OVERKIZ_TO_PRESET_MODE: dict[str, str] = {
+    OverkizCommandParam.GEOFENCING_MODE: PRESET_NONE,
+    OverkizCommandParam.SUDDEN_DROP_MODE: PRESET_NONE,
+    OverkizCommandParam.AWAY: PRESET_AWAY,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.FROSTPROTECTION: PRESET_FROST_PROTECTION,
+    OverkizCommandParam.MANUAL: PRESET_MANUAL,
+}
+PRESET_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODE.items()}
+
+TEMPERATURE_SENSOR_DEVICE_INDEX = 2
+
+
+class ValveHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Valve Heating Temperature Interface device."""
+
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_translation_key = DOMAIN
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            TEMPERATURE_SENSOR_DEVICE_INDEX
+        )
+
+        self._attr_min_temp = cast(
+            float, self.executor.select_state(OverkizState.CORE_MIN_SETPOINT)
+        )
+        self._attr_max_temp = cast(
+            float, self.executor.select_state(OverkizState.CORE_MAX_SETPOINT)
+        )
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation."""
+        return OVERKIZ_TO_HVAC_ACTION[
+            cast(str, self.executor.select_state(OverkizState.CORE_OPEN_CLOSED_VALVE))
+        ]
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature."""
+        return cast(
+            float, self.executor.select_state(OverkizState.CORE_TARGET_TEMPERATURE)
+        )
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        if temperature := self.temperature_device.states[OverkizState.CORE_TEMPERATURE]:
+            return temperature.value_as_float
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_DEROGATION,
+            float(temperature),
+            OverkizCommandParam.FURTHER_NOTICE,
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
```

Comment:
`hvac_mode` should be of type `HVACMode`
```suggestion
    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
```

It seems that pylint doesn't pick it up because we are inside `climate_entities` namespace, but I think it should still be implemented.